### ISSUE TITLE
Add --enabled-tools CLI argument to control exposed MCP tools

### ISSUE THREAD

**Issue by MarcusJellinghaus** [MarcusJellinghaus/mcp-code-checker#78]
## Summary

Add a `--enabled-tools` CLI argument that allows users to configure which MCP tools are exposed to clients. This enables more granular control over server functionality.

## Current Behavior

All 5 tools are always exposed to MCP clients:
- `run_pylint_check`
- `run_pytest_check`
- `run_mypy_check`
- `run_all_checks`
- `second_sleep`

## Proposed Behavior

Add `--enabled-tools` argument with comma-separated short names:

```bash
# All tools (default - backward compatible)
mcp-code-checker --project-dir .

# Only pylint and pytest
mcp-code-checker --project-dir . --enabled-tools pylint,pytest

# Explicit all tools
mcp-code-checker --project-dir . --enabled-tools "*"
```

## Tool Name Mapping

| Short name | Tool function |
|------------|---------------|
| `pylint` | `run_pylint_check` |
| `pytest` | `run_pytest_check` |
| `mypy` | `run_mypy_check` |
| `all_checks` | `run_all_checks` |
| `sleep` | `second_sleep` |

## Implementation Approach

**Changes to `main.py`:**
- Add `--enabled-tools` argument (comma-separated, optional)
- Default: `None` (means all tools enabled)
- Validate tool names against allowed list at startup
- Fail with error if invalid tool name provided (strict validation)

**Changes to `server.py`:**
- Add `enabled_tools: set[str] | None` parameter to `CodeCheckerServer.__init__()`
- Define tool name mapping as a constant
- In `_register_tools()`: only register tools that are in the enabled set (or all if `None`)

## Design Decisions

- **Default behavior**: All tools enabled (backward compatible)
- **Argument format**: Single comma-separated argument (`--enabled-tools pylint,pytest,mypy`)
- **Tool names**: Short names mapped internally (user-friendly)
- **Invalid names**: Fail at startup with error (strict validation)
- **Wildcard**: `"*"` means all tools (explicit alternative to omitting argument)

## Acceptance Criteria

- [ ] `--enabled-tools` argument added to CLI
- [ ] Default behavior unchanged (all tools exposed)
- [ ] Comma-separated tool names work correctly
- [ ] Invalid tool names cause startup failure with clear error message
- [ ] `"*"` enables all tools
- [ ] Only specified tools are exposed to MCP clients
- [ ] All existing tests pass
- [ ] New tests cover the filtering logic
